src/BlazorSignalR/BlazorSignalRExtensions.cs

```diff
@@ -33,15 +65,28 @@ public static class BlazorSignalRExtensions
             });
             if (options != null)
                 hubConnectionBuilder.Services.Configure(options);
-            hubConnectionBuilder.Services.AddSingleton(provider => BuildBlazorHttpConnectionFactory(provider, jsRuntime));
+            hubConnectionBuilder.Services.AddSingleton(provider => BuildBlazorHttpConnectionFactory(provider, jsRuntime, uriHelper));
             return hubConnectionBuilder;
         }
 
-        private static IConnectionFactory BuildBlazorHttpConnectionFactory(IServiceProvider provider, IJSRuntime jsRuntime)
+        private static IUriHelper GetUriHelper(IJSRuntime jSRuntime)
```

Comment:
I feel that as there is no way to get the UrlHelper when on server side, I think having this autodetection feature is only going to explode in users faces.

This should be removed and the user should always have to manually specify the UrlHelper, this way their code won't have to change between server side and client side.

Reply:
We have a breaking change then... We are using the UriHelper only for getting the document's base uri [here](https://github.com/AndreasTruetschel/BlazorSignalR/blob/master/src/BlazorSignalR/Internal/BlazorHttpConnection.cs#L129) and [here](https://github.com/AndreasTruetschel/BlazorSignalR/blob/master/src/BlazorSignalR/Internal/BlazorHttpConnection.cs#L360).  As we have a JSRuntime anyway we may replace the UriHelper parameter altogether with using `WebAssemblyUriHelper.Instance` and a custom js-interop as fallback. This is extremely simple, as the js part just [gets the documents base-uri](https://github.com/aspnet/AspNetCore/blob/master/src/Components/Browser.JS/src/Services/UriHelper.ts#L14).  This could then be encapsulated in a new type, say `BaseUriResolver` that handles all this. We don't have any breaking changes then.

Reply:
I'm not overly concerned about breaking changes ATM, as blazor itself is currently having major breaking changes, so people expect this.

While your solution is a good one, I'm not sure if we can justify all that extra code.

Reply:
I am not quite happy about this, but ok, I reverted these changes.